src/components/pages/home_screen/HomeScreen.tsx

```diff
@@ -0,0 +1,265 @@
+import React, { useEffect, useState } from "react";
+import { Grid, ThemeProvider, createTheme } from "@mui/material";
+import Sidebar from "./sidebar/Sidebar";
+import TaskList from "./task-list/TaskList";
+import TaskDetail from "./task-detail/TaskDetail";
+import { Task, Category } from "./types";
+import "./_homeScreen.scss";
+import Settings from "../setting_modal/Setting";
+import { db, auth } from "../../firebase";
+import {
+  collection,
+  query,
+  where,
+  getDocs,
+  addDoc,
+  deleteDoc,
+  doc,
+  updateDoc,
+} from "firebase/firestore";
+
+const theme = createTheme();
+
+interface HomeScreenProps {
+  onLogout: () => void;
+  userId: string;
+}
+
+const HomeScreen: React.FC<HomeScreenProps> = ({ onLogout, userId }) => {
+  const [isSidebarOpen, setIsSidebarOpen] = useState(true);
+  const [tasks, setTasks] = useState<Task[]>(() => {
+    const savedTasks = localStorage.getItem("tasks");
+    return savedTasks ? JSON.parse(savedTasks) : [];
+  });
+  const [categories, setCategories] = useState<Category[]>(() => {
+    const savedCategories = localStorage.getItem("categories");
+    return savedCategories ? JSON.parse(savedCategories) : [];
+  });
+  const [selectedTask, setSelectedTask] = useState<Task | null>(null);
+  const [selectedCategory, setSelectedCategory] = useState<string | null>(null);
+  const [isTaskDetailVisible, setIsTaskDetailVisible] = useState(false);
+  const [isSettingsVisible, setIsSettingsVisible] = useState(false);
+
+  const fetchTasksAndCategories = async () => {
+    try {
+      const currentUser = auth.currentUser;
```

Comment:
Dùng trực tiếp cách này luôn e, tương tự những chỗ khác đang xử lý giống vậy nhé
```suggestion
      const { currentUser } = auth;
```